neurom/features/morphology.py

```diff
@@ -374,18 +378,56 @@ def trunk_origin_radii(
         return [n.root_node.points[0][COLS.R]
                 for n in iter_neurites(morph, filt=is_type(neurite_type))]
 
-    def _mean_radius(points):
+    if min_length_filter is not None and min_length_filter <= 0:
+        raise NeuroMError(
+            "In 'trunk_origin_radii': the 'min_length_filter' value must be strictly greater "
+            "than 0."
+        )
+
+    if max_length_filter is not None and max_length_filter <= 0:
+        raise NeuroMError(
+            "In 'trunk_origin_radii': the 'max_length_filter' value must be strictly greater "
+            "than 0."
+        )
+
+    if (
+        min_length_filter is not None
+        and max_length_filter is not None
+        and min_length_filter >= max_length_filter
+    ):
+        raise NeuroMError(
+            "In 'trunk_origin_radii': the 'min_length_filter' value must be strictly less than the "
+            "'max_length_filter' value."
+        )
+
+    def _mean_radius(neurite):
+        points = neurite.root_node.points
         interval_lengths = morphmath.interval_lengths(points)
         path_lengths = np.insert(np.cumsum(interval_lengths), 0, 0)
         valid_pts = np.ones(len(path_lengths), dtype=bool)
         if min_length_filter is not None:
             valid_pts = (valid_pts & (path_lengths >= min_length_filter))
+            if not valid_pts.any():
+                warnings.warn(
+                    "In 'trunk_origin_radii': the 'min_length_filter' value is greater than the "
+                    "path distance of the last point of the last section so the radius of this "
+                    "point is returned."
+                )
+                return points[-1, COLS.R]
         if max_length_filter is not None:
-            valid_pts = (valid_pts & (path_lengths <= max_length_filter))
+            valid_max = (path_lengths <= max_length_filter)
+            valid_pts = (valid_pts & valid_max)
+            if not valid_pts.any():
+                warnings.warn(
+                    "In 'trunk_origin_radii': the 'min_length_filter' and 'max_length_filter' "
+                    "values excluded all the points of the section so the radius of the first "
+                    "point after the 'min_length_filter' path distance is returned."
+                )
+                # pylint: disable=invalid-unary-operand-type
+                return points[np.argwhere(~valid_max)[0][0], COLS.R]
```

Comment:
An alternative way to do this simpler would be:
return `points[~valid_max, COLS.R][0]`, right?

I sit guaranteed here that `np.argwhere(~valid_max)[0][0]` will return at least one value?

Reply:
Ah yes indeed we can just use a mask here.
The first element of `~valid_max` is always `True` because `max_length_filter > 0` so we are sure that at least the first point of the section is `True`.

Reply:
Cool. I don't have any other comments, so I am going to merge this. Thanks for your time!

Reply:
Ok, cool. Thanks for your review!